# Assignment 6 – HDS
**Name:** Marium Kalhoro 
**Course:** PUBH 1142 – Health Data Science  

## **Question 1: Use Python to calculate the following expressions**
3 × 8  
(33 + 7) / 5

In [4]:
# Question 1

print(3 * 8)            # 24
print((33 + 7) / 5)    # 8.0

24
8.0


## **Question 2: Determine the data type of 1+2 and 1+2. Explain why they are different**

In [5]:
# Question 2

print(type(1+2))
print(type(1+2.))

<class 'int'>
<class 'float'>


1+2 returns an integer (int), while 1+2. returns a floating point number (float).  

This is because 2. includes a decimal point, so Python treats it as a float. When an integer and a float are used together, Python converts the result to a float.

## **Question 3: Generate a loop that prints n² for n = {1, 2, 3, 4, 5}**

In [6]:
# Question 3

for n in range(1, 6):
    print(n**2)

1
4
9
16
25


## **Question 4: Create a loop that prints i² starting from i = 1 and stops when i² exceeds 1000. Print the final value of i and explain why i = 32**

In [7]:
# Question 4

i = 1

while i**2 < 1000:
    print(i**2)
    i += 1

print(i)

1
4
9
16
25
36
49
64
81
100
121
144
169
196
225
256
289
324
361
400
441
484
529
576
625
676
729
784
841
900
961
32


The final value of i is 32.

This is because:
31² = 961, which is still less than 1000  
32² = 1024, which is greater than 1000  

The loop stops when i² exceeds 1000, so the final value of i is 32.